src/tests/ftest/io/nvme_fragmentation.py

```diff
@@ -0,0 +1,122 @@
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+
+import time
+import threading
+from ior_test_base import IorTestBase
+from avocado.core.exceptions import TestFail
+
+try:
+    # python 3.x
+    import queue as queue
+except ImportError:
+    # python 2.7
+    import Queue as queue
+
+class NvmeFragmentation(IorTestBase):
+    # pylint: disable=too-many-ancestors
+    """Test class Description: Verify the drive fragmentation does free
+    the space and do not lead to ENOM_SPACE.
+
+    :avocado: recursive
+    """
+
+    def setUp(self):
+        """Set up for test case."""
+        super(NvmeFragmentation, self).setUp()
+        self.out_queue = queue.Queue()
+        self.write_flags = self.params.get("ior_flags", '/run/ior/iorflags/*')
+        self.apis = self.params.get("ior_api", '/run/ior/iorflags/*')
+        self.transfer_block_size = self.params.get("transfer_block_size",
+                                                   '/run/ior/iorflags/*')
+        self.obj_class = self.params.get("obj_class", '/run/ior/iorflags/*')
+
+    def run_ior_parallel(self, results):
+        """
+        IOR Thread function
+
+        results (queue): queue for returning thread results
+        """
+        container_info = {}
+        # Write IOR data for different transfer size
+        self.ior_cmd.flags.update(self.write_flags[0])
+        for oclass in self.obj_class:
+            self.ior_cmd.daos_oclass.update(oclass)
+            for api in self.apis:
+                self.ior_cmd.api.update(api)
+                for test in self.transfer_block_size:
+                    self.ior_cmd.transfer_size.update(test[0])
+                    self.ior_cmd.block_size.update(test[1])
+                    # run ior
+                    try:
+                        self.run_ior_with_pool()
+                    except  TestFail as error:
+                        print("--- FAIL --- IOR Command Failed {}"
+                              .format(error))
+                        results.put("FAIL")
+                    container_info[test[0]] = self.container.uuid
+
+        #Destroy the container created by thread
+        for key in container_info:
+            self.container.uuid = container_info[key]
+            self.container.destroy()
+
```

Comment:
This method seems to be good candidate to be used under ior_test_base.py (or) some generic test utils to spawn parallel jobs.

Reply:
There is one available in ior_test_base.py run_multiple_ior_with_pool() which runs limited job. Let me see if I can customize based on user input (How many process they want to run in parallel) and moved under ior_test_base.

Reply:
I tried to see if I can put this in common function but it seems those jobs are running the two threads only. It has linked with intercept library as well which does fuse operation.
This flow may not be the common for each scenarios so keeping it in test file for now.